### PR TITLE
Show full informational version for feature management library

### PR DESCRIPTION
**Pull request by amerjusupovic**

### CODE REVIEW

File: src/Microsoft.Extensions.Configuration.AzureAppConfiguration/TracingUtils.cs [hunk 1]
```diff
@@ -79,8 +80,27 @@ public static string GetAssemblyVersion(string assemblyName)
         {
             if (!string.IsNullOrEmpty(assemblyName))
             {
-                // Return the version using only the first 3 fields and remove additional characters
-                return AppDomain.CurrentDomain.GetAssemblies().SingleOrDefault(assembly => assembly.GetName().Name == assemblyName)?.GetName().Version?.ToString(3).Trim('{', '}');
+                AssemblyInformationalVersionAttribute[] infoVersionAttributes = AppDomain.CurrentDomain.GetAssemblies().SingleOrDefault(assembly => assembly.GetName().Name == assemblyName)?
+                    .GetCustomAttributes(typeof(AssemblyInformationalVersionAttribute), false) as AssemblyInformationalVersionAttribute[];
```
Comment (avanigupta) [inline src/Microsoft.Extensions.Configuration.AzureAppConfiguration/TracingUtils.cs:84]: Why dont we use `GetCustomAttribute<AssemblyInformationalVersionAttribute>()` instead?

Comment (amerjusupovic) [reply src/Microsoft.Extensions.Configuration.AzureAppConfiguration/TracingUtils.cs:84]: Not sure what happened but I tried using that method originally and I just didn't see it show up, I must've been calling in the wrong place or something. Silly mistake, let me fix it

File: src/Microsoft.Extensions.Configuration.AzureAppConfiguration/TracingUtils.cs [hunk 2]
```diff
@@ -79,8 +80,24 @@ public static string GetAssemblyVersion(string assemblyName)
         {
             if (!string.IsNullOrEmpty(assemblyName))
             {
-                // Return the version using only the first 3 fields and remove additional characters
-                return AppDomain.CurrentDomain.GetAssemblies().SingleOrDefault(assembly => assembly.GetName().Name == assemblyName)?.GetName().Version?.ToString(3).Trim('{', '}');
+                AssemblyInformationalVersionAttribute infoVersionAttribute = AppDomain.CurrentDomain.GetAssemblies().SingleOrDefault(assembly => assembly.GetName().Name == assemblyName)?
```
Comment (avanigupta) [inline src/Microsoft.Extensions.Configuration.AzureAppConfiguration/TracingUtils.cs:83]: This is getting too long. We could split it into two parts - get the assembly first. If it's not null, get the attribute and version.

File: src/Microsoft.Extensions.Configuration.AzureAppConfiguration/TracingUtils.cs [hunk 3]
```diff
@@ -79,8 +80,24 @@ public static string GetAssemblyVersion(string assemblyName)
...
+                    .GetCustomAttribute<AssemblyInformationalVersionAttribute>();
+                
+                string informationalVersion = infoVersionAttribute?.InformationalVersion;
+
+                if (string.IsNullOrEmpty(informationalVersion))
+                {
+                    return null;
+                }
+
+                int plusIndex = informationalVersion.IndexOf('+');
```
Comment (avanigupta) [inline src/Microsoft.Extensions.Configuration.AzureAppConfiguration/TracingUtils.cs:93]: could you add a short comment about why this is needed

**Review by avanigupta — APPROVED** [review]

**Pull request merged**